src/components/PriceRangePlot/PriceRangePlot.tsx

```diff
@@ -393,6 +395,85 @@ export const PriceRangePlot: React.FC<IPriceRangePlot> = ({
     disabled
   )
 
+  const heatmapLayer: Layer = ({ innerWidth, innerHeight }) => {
+    const heatmapData = [
+      {
+        v: 40000,
+        p1: -5,
+        p2: -1
+      },
+      {
+        v: 60000,
+        p1: -1,
+        p2: 1
+      },
+      {
+        v: 15000,
+        p1: 1,
+        p2: 2
+      },
+      {
+        v: 10000,
+        p1: 2,
+        p2: 4
+      },
+      {
+        v: 15000,
+        p1: 5,
+        p2: 10
+      }
+    ]
+    const maxHeight = innerHeight
+
+    const calculateConcentration = (v: number, p1: number, p2: number) => {
+      let priceRangeResult = p1 - p2
+      if (priceRangeResult >= 0) {
+        return v / priceRangeResult
+      } else {
+        return -(v / priceRangeResult)
+      }
+    }
+
+    const calculatedHeatmap = heatmapData.map(range => {
+      return {
+        ...range,
+        concentration: calculateConcentration(range.v, range.p1, range.p2)
+      }
+    })
+
+    return (
+      <>
+        {calculatedHeatmap.map((range, index) => {
+          return (
+            heatmapEnabled && (
+              <Tooltip
+                title={
+                  <>
+                    <Typography className={classes.heatmapValue}>{heatmapData[index].v}</Typography>
+                  </>
+                }
+                placement='top'
+                classes={{
+                  tooltip: classes.heatmapTooltip
+                }}>
+                <g>
+                  <rect
+                    x={range.p2 - plotMin}
+                    width={range.p2 - range.p1}
+                    key={index}
+                    y={innerHeight - maxHeight}
+                    height={maxHeight}
+                    fill={`rgba(46, 224, 154, 0.2)`}
```

Comment:
ranges should have different opacity values based on concentration value (they should be grouped in 5 categories, ranges in each one more or less transparent)